integration/upload_test.go

```diff
@@ -0,0 +1,257 @@
+// Copyright (c) 2022 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+
+//go:build integration
+
+package integration
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/fs"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/caarlos0/env/v6"
+	"github.com/eclipse-kanto/file-upload/client"
+	"github.com/eclipse-kanto/file-upload/uploaders"
+	"github.com/eclipse-kanto/kanto/integration/util"
+	"github.com/eclipse/ditto-clients-golang/protocol"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type uploadRequest struct {
+	CorrelationID string            `json:"correlationId"`
+	Options       map[string]string `json:"options"`
+}
+
+type uploadStatus struct {
+	State    string `json:"state"`
+	Progress int    `json:"progress"`
+}
+
+func (suite *fileUploadSuite) SetupSuite() {
+	suite.Setup(suite.T())
+
+	opts := env.Options{RequiredIfNoDef: false}
+	suite.uploadCfg = uploadTestConfig{}
+	require.NoError(suite.T(), env.Parse(&suite.uploadCfg, opts), "failed to process upload environment variables")
+	suite.T().Logf("upload test configuration - %v", suite.uploadCfg)
+	suite.checkUploadDir()
+
+	fmt.Println(suite.ThingCfg)
+	suite.thingURL = util.GetThingURL(suite.Cfg.DigitalTwinAPIAddress, suite.ThingCfg.DeviceID)
+	suite.featureURL = util.GetFeatureURL(suite.thingURL, featureID)
+}
+
+func (suite *fileUploadSuite) TearDownSuite() {
+	suite.TearDown()
+}
+
+func TestFileUploadHTTP(t *testing.T) {
```

Comment:
I guess that test methods can follow the suite names. It's more readable i.e. this one can be: TestHTTPFileUpload